### PR TITLE
Add signup page

### PR DESCRIPTION
**Pull request by naomiajacobs**
Hooks yo!!!!!!! ☠️ ↪️ ↩️ ☠️ 

Adds
- signup page
- logout page
- prettier formatting for frontend code
- starter for integration tests
  - still not done, needs db setup

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -1,32 +1,49 @@
...
 # OMDB API Key
+
 Vist http://www.omdbapi.com/apikey.aspx to generate an OMDB API key. Add a `.env` file to the root, with contents of
+
 ```
 OMDB_API_KEY=YOUR_KEY
 ```
 
 ########FRONTEND########
+
```
Comment (norseboar) [inline SETUP.md]: Are all these spaces your linter? Just curious

File: app/controllers/signup_controller.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from app.models import User
+
+from flask import jsonify, request
+
+from app.session_util import create_session
+
+
+class SignupController():
+    def handle(self):
+        data = json.loads(request.data)
```
Comment (norseboar) [inline app/controllers/signup_controller.py]: I think `request.json` will work if the front-end is sending the data back w/ `application/json`

File: app/routes.py [hunk 3]
```diff
@@ -18,6 +14,16 @@ def serve(path):
     return send_from_directory(BUILD_DIR, 'index.html')
 
 
+@app.route('/current-user')
```
Comment (norseboar) [inline app/routes.py]: Can we get rid of this API now that we have the session?

File: app/session_util.py [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
 def delete_session():
     session.clear()
 
+
 def create_session(email):
     session['email'] = email
 
+
+def get_current_session_user():
+    return User.query.filter(User.email == session['email']).first()
```
Comment (norseboar) [inline app/session_util.py]: Can use `one_or_none` here for a bit more specificity/error-catching

File: integration-tests/sessions_test.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        self.assertEqual(driver.title, "Kvasir Movies")
+        driver.find_element_by_class_name("signup").click()
+
+        email_input = driver.find_element_by_name('email')
+        email_input.clear()
+        email_input.send_keys('naomi@jacobs.com')
+        password_input = driver.find_element_by_name('password')
+        password_input.clear()
+        password_input.send_keys('super secret')
+        driver.close()
```
Comment (norseboar) [inline integration-tests/sessions_test.py]: I might just not be familiar w/ the framework, but what's the rest of the test for w/o assertions? See if anything raises a python-side error?

File: src/pages/HomePage.tsx [hunk 6]
```diff
@@ -1,39 +1,59 @@
...
-            <p>
-                {this.state.sessionInfo && this.state.sessionInfo.is_session_active ? `Welcome back, ${this.state.sessionInfo.email}! ` : ''}
-                Find 🎬 with 👫 :D
-            </p>
-            <a href='/login'>Log In</a>
-            </header>
-        </div>
-        );
-    }
+  readonly state: { sessionInfo: SessionInfo | null } = { sessionInfo: null };
```
Comment (norseboar) [inline src/pages/HomePage.tsx]: Did this whole thing get unindented by two spaces? Is this a lint thing?

File: src/pages/LoginPage.tsx [hunk 7]
```diff
@@ -1,55 +1,66 @@
...
-        email: string | null,
-        password: string | null,
-        sessionInfo: SessionInfo | null,
-    } = {
-        email: null,
-        password: null,
-        sessionInfo: null,
-    };
+  readonly state: {
+    email: string | null;
```
Comment (norseboar) [inline src/pages/LoginPage.tsx]: in ts are you supposed to terminate w/ `;`s in a type definition? Don't think I've seen a mix of the syntaxes before

File: SETUP.md (hunk 1 above)
Comment (naomiajacobs) [reply SETUP.md]: Oooooh yeah probably; I told `prettier` to apply to markdown too

File: app/controllers/signup_controller.py (hunk 2 above)
Comment (naomiajacobs) [reply app/controllers/signup_controller.py]: 😮 oooh awesome!!!

File: integration-tests/sessions_test.py (hunk 5 above)
Comment (naomiajacobs) [reply integration-tests/sessions_test.py]: Yeah, before we have a test db set up, I don't want to actually run the test cause it'll add data to my db. Ideally, we would go through the actual entire signup flow, but until then, this tests that
- the home page renders
- clicking signup brings you to the signup page
- both inputs exist and are type-able

It's not super useful right now, but once I get the rest of the test scaffolding set up, it'll be more fleshed out. This was mostly meant to be a functional example to make sure the driver worked and to get it integrated with travisCI

File: src/pages/HomePage.tsx (hunk 6 above)
Comment (naomiajacobs) [reply src/pages/HomePage.tsx]: Yep, my editor was displaying 2 spaces but was committing 4 spaces for some reason; `prettier` fixed that

File: src/pages/LoginPage.tsx (hunk 7 above)
Comment (naomiajacobs) [reply src/pages/LoginPage.tsx]: ya, type and interface definitions are terminated with semicolons - it's like doing
```
interface LoginState {
  email: string | null;
  password: string | null;
}

class LoginPage extends Component {
  readonly state: LoginState = { ... }
  ...
```

File: integration-tests/sessions_test.py (hunk 5 above)
Comment (norseboar) [reply integration-tests/sessions_test.py]: Sounds good. So IIUC, we never need to assert anything, b/c this is checking that nothing throws an error while on this path?

**Pull request merged**